component/src/test/java/io/siddhi/extension/execution/string/FillTemplateFunctionExtensionTestCase.java

```diff
@@ -152,8 +153,123 @@ public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
         siddhiAppRuntime.shutdown();
     }
 
-    @Test(expectedExceptions = SiddhiAppCreationException.class)
+    @Test
     public void testFillTemplate3() throws InterruptedException {
+        LOGGER.info("FillTemplateFunctionExtension TestCase, where the argument is a valid map");
+        SiddhiManager siddhiManager = new SiddhiManager();
+        siddhiManager.setExtension("map:create", CreateFunctionExtension.class);
+
+        String inStreamDefinition = "define stream inputStream (template string, symbol string, volume string, " +
+                "price string);";
+
+        String template = "The stock price of {{symbol}}. Volume of {{symbol}} is " +
+                "{{volume}} and the stock price is {{price}}";
+        String query = (
+                "@info(name = 'query1') from inputStream select " +
+                        "str:fillTemplate(template, " +
+                        "map:create('symbol', symbol, 'volume', volume, 'price', price)) as msg " +
+                        "insert into outputStream;"
+        );
+
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(inStreamDefinition + query);
+
+        siddhiAppRuntime.addCallback("query1", new QueryCallback() {
+            @Override
+            public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+                EventPrinter.print(timeStamp, inEvents, removeEvents);
+                for (Event event : inEvents) {
+                    count.incrementAndGet();
+                    if (count.get() == 1) {
+                        AssertJUnit.assertEquals("The stock price of WSO2. Volume of WSO2 is 100 and the stock price " +
+                                "is " + "111.11", event.getData(0));
+                        eventArrived = true;
+                    }
+                    if (count.get() == 2) {
+                        AssertJUnit.assertEquals("The stock price of IBM. Volume of IBM is 200 and the stock price " +
+                                "is " + "222.22", event.getData(0));
+                        eventArrived = true;
```

Comment:
You can move eventArrived = true to outside

Reply:
Fixed in https://github.com/siddhi-io/siddhi-execution-string/pull/70/commits/dcaa870e2373b6082913f67285a7438dde477b7e